### ISSUE TITLE
missing a shared file led to massive log spamming with a possible fix

### ISSUE THREAD

**Issue by mightyBroccoli** [nextcloud/server#3492]
### Steps to reproduce
1. share some file to a userA
2. for whatever reason the file gets deleted/moved eg from the filesystem
3. userA 's client will try to get this file from the server as the share is still valid
4. do the usuall stuff occ files:scan --all  and occ maintenance:repair to fix this issue
5. share is still active and the file is not acceable via the webui to even delete the share to userA

### wanted behavior
1. let occ maintenance:repair check all shares if they are still valid if not drop the share to get around this problem

### Server configuration
**Operating system**: Linux 3.16.0-4-amd64 #1 SMP Debian 3.16.39-1 (2016-12-30) x86_64

**Web server:** Apache 2.4.25

**Database:** mysql 5.5.54

**PHP version:** 7.0.15-1~dotdeb+8.1
Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, Reflection, SPL, session, standard, apache2handler, igbinary, msgpack, mysqlnd, PDO, xml, apcu, calendar, ctype, curl, dom, mbstring, fileinfo, ftp, gd, gettext, iconv, imagick, intl, json, exif, mcrypt, memcached, mysqli, pdo_mysql, Phar, posix, readline, redis, shmop, SimpleXML, sockets, sysvmsg, sysvsem, sysvshm, tokenizer, wddx, xmlreader, xmlwriter, xsl, zip, Zend OPcache

**Nextcloud version:** 11.0.1 (stable) - 11.0.1.2

**Updated from an older Nextcloud/ownCloud or fresh install:**
updated all the way from owncloud 9

**Where did you install Nextcloud from:** 
the official website 

**List of activated apps:**
<details>
	<summary>App list</summary>
```
Enabled:
 - activity: 2.4.1
 - admin_audit: 1.1.0
 - announcementcenter: 3.0.0
 - apporder: 0.3.3
 - audioplayer: 1.4.1
 - calendar: 1.5.0
 - comments: 1.1.0
 - contacts: 1.5.2
 - dav: 1.1.1
 - direct_menu: 0.10.0
 - federatedfilesharing: 1.1.1
 - federation: 1.1.1
 - files: 1.6.1
 - files_automatedtagging: 1.1.1
 - files_downloadactivity: 1.0.0
 - files_external: 1.1.2
 - files_pdfviewer: 1.0.1
 - files_retention: 1.0.1
 - files_sharing: 1.1.1
 - files_texteditor: 2.2
 - files_trashbin: 1.1.0
 - files_versions: 1.4.0
 - files_videoplayer: 1.0.0
 - firstrunwizard: 2.0
 - gallery: 16.0.0
 - issuetemplate: 0.2.0
 - logreader: 2.0.0
 - lookup_server_connector: 1.0.0
 - news: 10.1.0
 - nextcloud_announcements: 1.0
 - notifications: 1.0.1
 - password_policy: 1.1.0
 - previewgenerator: 1.0.1
 - provisioning_api: 1.1.0
 - serverinfo: 1.1.1
 - sharebymail: 1.0.1
 - systemtags: 1.1.3
 - templateeditor: 0.2
 - theming: 1.1.1
 - twofactor_backupcodes: 1.0.0
 - twofactor_totp: 1.0.0
 - updatenotification: 1.1.1
 - workflowengine: 1.1.1

Disabled:
 - encryption
 - external
 - files_accesscontrol
 - survey_client
 - user_external
 - user_ldap
 - user_saml
```
</details>

**The content of config/config.php:**
<details>
	<summary>Config report</summary>

```
{
    "instanceid": "ocdwzxwi729i",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "cloud.magicbroccoli.de"
    ],
    "datadirectory": "\/var\/www\/magicbroccoli.de\/data",
    "version": "11.0.1.2",
    "installed": true,
    "cipher": "AES-256-CFB",
    "dbtype": "mysql",
    "dbname": "nextcloud",
    "dbhost": "localhost",
    "dbtableprefix": "oc_",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "mysql.utf8mb4": true,
    "default_language": "de",
    "defaultapp": "files",
    "enable_avatars": true,
    "allow_user_to_change_display_name": true,
    "remember_login_cookie_lifetime": 864000,
    "session_lifetime": 86400,
    "auth.bruteforce.protection.enabled": true,
    "updatechecker": true,
    "updater.server.url": "https:\/\/updates.nextcloud.com\/updater_server\/",
    "updater.release.channel": "stable",
    "upgrade.disable-web": false,
    "has_internet_connection": true,
    "check_for_working_htaccess": true,
    "check_for_working_webdav": true,
    "check_for_working_wellknown_setup": true,
    "appstoreenabled": true,
    "appstoreurl": "https:\/\/apps.nextcloud.com\/api\/v0",
    "appstore.experimental.enabled": true,
    "appcodechecker": true,
    "memcache.local": "\\OC\\Memcache\\APCu",
    "filelocking.enabled": "true",
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "redis": {
        "host": "\/var\/run\/redis\/redis.sock",
        "port": 0,
        "timeout": 0
    },
    "overwrite.cli.url": "https:\/\/cloud.magicbroccoli.de",
    "htaccess.RewriteBase": "\/",
    "trashbin_retention_obligation": "auto, 60",
    "versions_retention_obligation": "auto",
    "activity_expire_days": 80,
    "filesystem_check_changes": 0,
    "theme": "",
    "maintenance": false,
    "singleuser": false,
    "data-fingerprint": "16b42c01070fc24c2086fb974eda1d16",
    "mail_smtpmode": "smtp",
    "mail_from_address": "nextcloud",
    "mail_domain": "magicbroccoli.de",
    "mail_smtpauth": 1,
    "mail_smtpauthtype": "LOGIN",
    "mail_smtphost": "mail.server-speed.net",
    "mail_smtpport": "587",
    "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
    "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpsecure": "tls",
    "logdateformat": "d.m.Y - H:i:s",
    "logtimezone": "Europe\/Berlin",
    "log_rotate_size": 52428800,
    "loglevel": 2,
    "logfile": "\/var\/log\/nextcloud\/nextcloud.log",
    "enable_previews": true,
    "preview_max_x": 1024,
    "preview_max_y": 1024,
    "preview_max_filesize_image": 25,
    "preview_libreoffice_path": "\/usr\/bin\/libreoffice",
    "preview_office_cl_parameters": " --headless --nologo --nofirststartwizard --invisible --norestore --convert-to pdf --outdir "
}
```
</details>

**Are you using external storage, if yes which one:** Array
(
    [1] => \OC\Files\Storage\DAV
    [2] => \OCA\Files_External\Lib\Storage\OwnCloud
    [3] => \OCA\Files_External\Lib\Storage\Dropbox
    [4] => \OCA\Files_External\Lib\Storage\Google

)

**Are you using encryption:** no

#### Nextcloud log (data/nextcloud.log)
<details>
	<summary>Nextcloud log</summary>
	```
{"reqId":"2J0UMrgDbZ+cfEEMP4jJ","remoteAddr":"x.x.x.x","app":"webdav","message":"Exception: {\"Message\":\"HTTP\\\/1.1 503 Failed to open file: File with id \\\"274188\\\" has not been found.\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\ServiceUnavailable\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/domain.org\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(85): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->get()\\n#1 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpGet(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#2 \\\/var\\\/www\\\/domain.org\\\/nextcloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#3 \\\/var\\\/www\\\/domain.org\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(479): Sabre\\\\Event\\\\EventEmitter->emit('method:GET', Array)\\n#4 \\\/var\\\/www\\\/domain.org\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#5 \\\/var\\\/www\\\/domain.org\\\/nextcloud\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/webdav.php(60): Sabre\\\\DAV\\\\Server->exec()\\n#6 \\\/var\\\/www\\\/domain.org\\\/nextcloud\\\/remote.php(165): require_once('\\\/var\\\/www\\\/magicb...')\\n#7 {main}\",\"File\":\"\\\/var\\\/www\\\/domain.org\\\/nextcloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/File.php\",\"Line\":314,\"User\":\"joerg\"}","level":4,"time":"13.02.2017 - 12:30:09","method":"GET","url":"\/remote.php\/webdav\/data\/textfile.txt","user":"user","version":"11.0.1.2"}
	```
</details>

**Comment by nickvergessen** [comment]
I'm not sure there is a lot we can and should do against this. It is very valid to create a log entry in that case. The repair step that checks shares just cross checks with the filecache.

So if something should change here at all, it should be the files:scan command, to not only detect new ones, but also delete old ones.

**Comment by mightyBroccoli** [comment]
Are there any plans to implement such a feature? It would be really helpful to just check this easily ans repair it easily.

**Comment by inducer** [comment]
I have this on my instance, and it's really not great. I've shared some photos with my father ("heinz"). Due to some glitch, it appears that some files were no longer found under their original location, leading the `occ` `files:scan` command to recreate them. As a result, I had to recreate all my shares. So far so good, however this left my dad with a few shares that

* are not removable, with the following backtrace:

```
{"reqId":"JMWn0QSeziFG4vVXNciY","level":3,"time":"2018-07-17T20:56:32+00:00","remoteAddr":"79.234.93.93","user":"heinz","app":"no app in context","method":"GET","url":"\/nextcloud\/ocs\/v2.php\/apps\/files_sharing\/api\/v1\/shares?format=json&path=%2Fpuerto-rico&reshares=true","message":"Exception: {\"Exception\":\"OCP\\\\Files\\\\StorageNotAvailableException\",\"Message\":\"File with id \\\"52195\\\" has not been found.\",\"Code\":0,\"Trace\":\"#0 \\\/web\\\/nextcloud-13.0.0\\\/apps\\\/files_sharing\\\/lib\\\/SharedStorage.php(405): OC\\\\Files\\\\Storage\\\\FailedStorage->acquireLock('', 1, Object(OC\\\\Lock\\\\MemcacheLockingProvider))\\n#1 \\\/web\\\/nextcloud-13.0.0\\\/lib\\\/private\\\/Files\\\/Storage\\\/Wrapper\\\/Wrapper.php(592): OCA\\\\Files_Sharing\\\\SharedStorage->acquireLock('', 1, Object(OC\\\\Lock\\\\MemcacheLockingProvider))\\n#2 \\\/web\\\/nextcloud-13.0.0\\\/lib\\\/private\\\/Files\\\/View.php(1926): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Wrapper->acquireLock('', 1, Object(OC\\\\Lock\\\\MemcacheLockingProvider))\\n#3 \\\/web\\\/nextcloud-13.0.0\\\/lib\\\/private\\\/Files\\\/View.php(2036): OC\\\\Files\\\\View->lockPath('\\\/heinz\\\/files\\\/pu...', 1, false)\\n#4 \\\/web\\\/nextcloud-13.0.0\\\/lib\\\/private\\\/Files\\\/Node\\\/Node.php(360): OC\\\\Files\\\\View->lockFile('\\\/heinz\\\/files\\\/pu...', 1)\\n#5 \\\/web\\\/nextcloud-13.0.0\\\/apps\\\/files_sharing\\\/lib\\\/Controller\\\/ShareAPIController.php(936): OC\\\\Files\\\\Node\\\\Node->lock(1)\\n#6 \\\/web\\\/nextcloud-13.0.0\\\/apps\\\/files_sharing\\\/lib\\\/Controller\\\/ShareAPIController.php(596): OCA\\\\Files_Sharing\\\\Controller\\\\ShareAPIController->lock(Object(OC\\\\Files\\\\Node\\\\Folder))\\n#7 [internal function]: OCA\\\\Files_Sharing\\\\Controller\\\\ShareAPIController->getShares('false', 'true', 'false', Object(OC\\\\Files\\\\Node\\\\Folder), 'false')\\n#8 \\\/web\\\/nextcloud-13.0.0\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(161): call_user_func_array(Array, Array)\\n#9 \\\/web\\\/nextcloud-13.0.0\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(91): OC\\\\AppFramework\\\\Http\\\\Dispatcher->executeController(Object(OCA\\\\Files_Sharing\\\\Controller\\\\ShareAPIController), 'getShares')\\n#10 \\\/web\\\/nextcloud-13.0.0\\\/lib\\\/private\\\/AppFramework\\\/App.php(115): OC\\\\AppFramework\\\\Http\\\\Dispatcher->dispatch(Object(OCA\\\\Files_Sharing\\\\Controller\\\\ShareAPIController), 'getShares')\\n#11 \\\/web\\\/nextcloud-13.0.0\\\/lib\\\/private\\\/AppFramework\\\/Routing\\\/RouteActionHandler.php(47): OC\\\\AppFramework\\\\App::main('OCA\\\\\\\\Files_Shari...', 'getShares', Object(OC\\\\AppFramework\\\\DependencyInjection\\\\DIContainer), Array)\\n#12 [internal function]: OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler->__invoke(Array)\\n#13 \\\/web\\\/nextcloud-13.0.0\\\/lib\\\/private\\\/Route\\\/Router.php(297): call_user_func(Object(OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler), Array)\\n#14 \\\/web\\\/nextcloud-13.0.0\\\/ocs\\\/v1.php(78): OC\\\\Route\\\\Router->match('\\\/ocsapp\\\/apps\\\/fi...')\\n#15 \\\/web\\\/nextcloud-13.0.0\\\/ocs\\\/v2.php(24): require_once('\\\/web\\\/nextcloud-...')\\n#16 {main}\",\"File\":\"\\\/web\\\/nextcloud-13.0.0\\\/lib\\\/private\\\/Files\\\/Storage\\\/FailedStorage.php\",\"Line\":197,\"Hint\":\"Storage is temporarily not available\"}","userAgent":"Mozilla\/5.0 (X11; Linux i686; rv:52.0) Gecko\/20100101 Firefox\/52.0","version":"13.0.0.14"}
```
* cause errors ("storage not available") when you try to browse them,
* crash the sync client when it encounters them.

So what you've got here is a reliably triggerable way to (for now) irrecoverably turn your nextcloud instance into the IT equivalent of a paperweight.
